tests/ui/target-feature/wasm-relaxed-simd.rs

```diff
@@ -0,0 +1,9 @@
+//@ only-wasm32-wasip1
+//@ compile-flags: -Ctarget-feature=+relaxed-simd,+simd128 --crate-type=lib
```

Comment:
`v128` can't really be used without enabling `simd128`, so the latest CI failure makes sense (I fixed it now as can be seen here).
I'm unaware if there is a mechanism that allows target features to implicitly enable other target features.

Reply:
I've experimentally implemented a dependant target feature mechanism in https://github.com/daxpedda/rust/compare/wasm-relaxed-simd...daxpedda:rust:dependant-target-features (still missing the GCC implementation and adjusting Stdarch), let me know if this looks like a good idea and I can add it to this PR.

Reply:
I'm not able to reproduce the CI failure locally so I'm not 100% sure what's going on. I wouldn't recommend going the route of adding dependencies between features, I think it's reasonable to update stdarch to enable `simd128` wherever `relaxed-simd` is also enabled.

The `-Ctarget-feature` flag should never be required to get intrinsics to work correctly, they should always work without it.

Reply:
Ah ok I was able to reproduce this locally. Reproduction requires the changes in this PR to use the new `llvm.wasm.relaxed.*` intrinsics.

I was able to fix this with https://github.com/rust-lang/stdarch/pull/1610

Reply:
> The `-Ctarget-feature` flag should never be required to get intrinsics to work correctly, they should always work without it.

The mechanism I added in https://github.com/daxpedda/rust/compare/wasm-relaxed-simd...daxpedda:rust:dependant-target-features makes it so that using `-Ctarget-feature=relaxed-simd` requires also using `-Ctarget-feature=simd128`. It's unrelated to `#[enable = ...]`, which you fixed in https://github.com/rust-lang/stdarch/pull/161, thanks!

But setting `-Ctarget-feature` correctly here is still important to mark the Wasm module required proposals section, which Wasm engines rely on when loading the module (unless `#[enable = ...]` does this as well? but I think not).

Not sure if https://github.com/llvm/llvm-project/issues/98502 covers this or not (thank you for filing and moving it along!).

Reply:
Ah ok I should have read your patch more closely! Personally though I'd advocate for the opposite of an error, so when `relaxed-simd` is specified it automatically enables `simd128`. Given the direction https://github.com/llvm/llvm-project/issues/98502 is going I suspect that LLVM won't change here, so I think it'll become our responsibility as a frontend to enable this.

Reply:
I will look into it then.

Reply:
Done.
Both `-Ctarget-feature=+relaxed-simd` and `#[target_feature(enable = "relaxed-simd")]` should implicitly enable `simd128`.

Reply:
Note that the Stdarch patch was reverted again in https://github.com/rust-lang/stdarch/pull/1612, as its now not necessary anymore.